wavefront/wavefront.go

```diff
@@ -51,20 +55,70 @@ type (
 		listener            net.Listener
 		server              *grpc.Server
 		reporterInitialized bool
+		reporter            wf.WavefrontMetricsReporter
 	}
 )
 
 // ensure that WavefrontAdapter implements the HandleMetricServiceServer interface.
 var _ metric.HandleMetricServiceServer = &WavefrontAdapter{}
 
 // createWavefrontReporter creates a reporter that periodically flushes metrics to Wavefront.
-func createWavefrontReporter(cfg *config.Params) {
+func (wa *WavefrontAdapter) createWavefrontReporter(cfg *config.Params) {
 	hostTags := map[string]string{"source": cfg.Source}
+
```

Comment:
Check if  WavefrontAdapter.reporterInitialized is false before proceeding to create a new reporter. I don't see that parameter being used elsewhere (please verify this).
Alternatively, can also check if WavefrontAdapter.reporter == nil , before creating a new one. (Also remove wa.reporterInitialized if its not being used)

Reply:
The bool flag is removed and the condition check is now on WavefrontAdapter.reporter == nil